pontoon/base/static/js/translate.js

```diff
@@ -600,10 +600,11 @@ var Pontoon = (function (my) {
      * callback Callback function
      */
     checkUnsavedChanges: function (callback) {
-      var entity = this.getEditorEntity();
+      var entity = this.getEditorEntity(),
+	      self = this;
 
       // Ignore for anonymous users, for which we don't save translations
-      if (!this.user.email || !entity) {
+      if (!self.user.uid || !entity) {
         this.restoreInPlaceTranslation();
         return callback();
       }
```

Comment:
Correct, not the entire block - we should only remove the `!self.user.uid || ` part.